MaxText/layers/attention_op.py

```
@@ -1065,8 +1071,13 @@ def wrap_flash_attention(
           decoder_segment_ids_tuple = splash_attention_kernel.SegmentIds(decoder_segment_ids_q, decoder_segment_ids_kv)
       else:
         decoder_segment_ids_tuple = None
-      attention_output = jax.vmap(splash_kernel)(query, key, value, segment_ids=decoder_segment_ids_tuple)
-
+      # TODO(ranran): remove if/else branch once b/441336842 is fixed
+      if version.parse(jax.__version__) < version.parse("0.7.2.dev20250824"):
+        attention_output = jax.vmap(splash_kernel)(query, key, value, decoder_segment_ids_tuple)
```

This seems to silently ignore attention sink. Shall we add validation in pyconfig when using attention_sink and flash attention?
```
if version.parse(jax.__version__) < version.parse("0.7.2.dev20250824") and raw_key["attention_sink"] and raw_key["attention"] == "flash":
  raise ValueError(...)
```

---

Most of tests will fail if we add this validation. Currently, the github runner are using 0.7.0 JAX version in the candidate image, and the flash attention feature with sinks is only supports starting 0.7.2 dev. That's the reason we have this branch here :)